sympy/ntheory/modular.py

```diff
@@ -253,3 +253,40 @@ def combine(c1, c2):
         if symmetric:
             return symmetric_residue(n, m), m
         return n, m
+
+
+def crt_cartesian(rem, mod):
+    """
+    Return a list of values whose elements give the remainders
+    in the Cartesian product of the supplied remainders when divided
+    by each of the supplied moduli. None is returned if the moduli
+    are not pairwise-prime.
```

Comment:
It seems better to raise an exception in this case.

Reply:
It would help to spell this out as a formula.

Reply:
@asmeurer , if we have to raise an exception, then we can do something like below:
```
        try:
            mi = mod_inverse(m, mod[i])*m
        except:
            raise ValueError("two or more moduli aren't co-prime")  # mod_inverse fails if values are not coprime
```

Reply:
Also, the test cases will have to be updated accordingly

Reply:
@smichr said that we can return None when any pair of moduli aren't co-prime like done in ```crt``` function. So what should we do? Either update both ```crt``` and ```crt_cartesian``` to raise ValueError or just return None as is in both the functions

Reply:
I would just let the exception from mod_inverse bubble up. 

I didn't realize that crt already returns None. Returning a different type in the case of error is not very Pythonic. It's better to raise an exception. Otherwise the user has to check the type, and they also don't get any sort of error message.

Reply:
Ok so should I include a ValueError exception in ```crt``` as well?

Reply:
@asmeurer So for ValueError exception, we can add a function in ```modular.py``` that takes in two parameters namely ```mods``` which is list of moduli and a moduli ```m``` when a ValueError exception has been raised by the routine.

Now this function which takes ```mods``` and ```m``` as it's arguments will return a pair of moduli which aren't co-prime and in this pair, one of the moduli will be m. This pair can be printed as an output so that the user gets a sense of where the program is going wrong.

This function can be used by both the ```crt``` and ```crt_cartesian``` functions.

Reply:
@asmeurer When I was testing ```crt``` function, I found out that when any moduli is ```1```, the function still returns an output though it is supposed to be not correct and should raise a ValueError exception IMO.